### ISSUE TITLE
Consider not warning on `export *` when re-exported file is passing

### ISSUE THREAD

**Issue by silverwind** [ArnaudBarre/eslint-plugin-react-refresh#24]
I think a common pattern in a library is to re-export everything in a `index.jsx` for convenience, for example:

```js
export * from "./Buttons.jsx";
```

Currently, this plugin will error on every such line even thought the re-exported files themselves pass. Is this really an issue with the HMR or a linter bug?

**Comment by ArnaudBarre** [comment]
For me the issue is using `.jsx` for a file that is just re-exporting stuff

**Comment by silverwind** [comment]
Ah I see [the problem](https://github.com/silverwind/eslint-config-silverwind-react/blob/b8cfa841fbdfac05bd6671e979593cad47319a42/.eslintrc.yaml#L45) in my config. ``react/jsx-filename-extension` is set to `always` but should be set to `as-needed`, in which case is will error when a `.jsx` file contains non-jsx, after which I can turn off `checkJS ` an both rules will be happy.

I still wonder, why is `export *` so problematic for this plugin? Isn't this export pretty much a noop pass-through forward to the imported file?

**Comment by silverwind** [comment]
Within the library, I don't use those `*` but only direct imports, so I guess I am fine in regards to HMR there.

When using the lib in an application and maybe symlinking the lib into the application's `node_modules`, I guess I will still be affected, so I'll evaluate whether it's worth the convenience to not have working HMR while symlinked. Thanks for your time.
